cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -359,31 +360,41 @@ public ClusterOperationMetadata createCluster(String name, String imageVersion,
                         Boolean.toString(conf.isStackdriverMonitoringEnabled()));
 
 
+      ClusterConfig.Builder builder = ClusterConfig.newBuilder()
+        .setMasterConfig(InstanceGroupConfig.newBuilder()
+                           .setNumInstances(conf.getMasterNumNodes())
+                           .setMachineTypeUri(conf.getMasterMachineType())
+                           .setDiskConfig(DiskConfig.newBuilder()
+                                            .setBootDiskSizeGb(conf.getMasterDiskGB())
+                                            .setNumLocalSsds(0)
+                                            .build())
+                           .build())
+        .setWorkerConfig(InstanceGroupConfig.newBuilder()
+                           .setNumInstances(conf.getWorkerNumNodes())
+                           .setMachineTypeUri(conf.getWorkerMachineType())
+                           .setDiskConfig(DiskConfig.newBuilder()
+                                            .setBootDiskSizeGb(conf.getWorkerDiskGB())
+                                            .setNumLocalSsds(0)
+                                            .build())
+                           .build())
+        .setGceClusterConfig(clusterConfig.build())
+        .setSoftwareConfig(SoftwareConfig.newBuilder()
+                             .setImageVersion(imageVersion)
+                             .putAllProperties(dataprocProps));
+
+      if (conf.getGcpCmekKeyName() != null) {
+        builder.setEncryptionConfig(EncryptionConfig.newBuilder()
+                                      .setGcePdKmsKeyName(conf.getGcpCmekKeyName()).build());
+      }
+
+      if (conf.getGcpCmekBucket() != null) {
```

Comment:
I think custom GCS bucket is actually not specific for CMEK use case only. Even you use google encryption key, you can still provide a customer GCS bucket. I don't think they are tied together as shown in the GCP console when creating Dataproc cluster.